include/fi_util.h

```diff
@@ -339,6 +340,32 @@ struct util_event {
 int ofi_eq_create(struct fid_fabric *fabric, struct fi_eq_attr *attr,
 		 struct fid_eq **eq_fid, void *context);
 
+/*
+ * MR
+ */
+
+
+typedef void *util_mr_itr;
+/*hide addr related info & store prov_mr ptr */
+typedef struct fi_mr_attr util_mr_item_t;
+
+struct util_mr {
+    void *map_handle;
+    uint64_t b_key; /* track available key (BASIC usage) */
+    enum fi_mr_mode mr_type;
+};
+
+extern struct util_mr * ofi_mr_init(enum fi_mr_mode mode);
+extern int ofi_mr_insert(struct util_mr * in_mr_h,
+                                const struct fi_mr_attr *in_attr,
+                                uint64_t * out_key, void * in_prov_mr);
```

Comment:
Can you add some usage info here? Like what is the use of *in/out_prov_mr etc. That will be useful for other provider implementors.